Sources/App/Scenes/CarPlaySceneDelegate.swift

```diff
@@ -0,0 +1,261 @@
+    func unsubscribe()
+}
+
+@available(iOS 16.0, *)
+class CarPlayDelegate: UIResponder {
+    public static let SUPPORTED_DOMAINS_WITH_STRING = [
+        "button": L10n.Carplay.Labels.buttons,
+        "cover": L10n.Carplay.Labels.covers,
+        "input_boolean": L10n.Carplay.Labels.inputBooleans,
+        "input_button": L10n.Carplay.Labels.inputButtons,
+        "light": L10n.Carplay.Labels.lights,
+        "lock": L10n.Carplay.Labels.locks,
+        "scene": L10n.Carplay.Labels.scenes,
+        "script": L10n.Carplay.Labels.scripts,
+        "switch": L10n.Carplay.Labels.switches
+    ]
+    
+    public let SUPPORTED_DOMAINS = SUPPORTED_DOMAINS_WITH_STRING.keys
+    
+    private var MAP_DOMAINS = [
+        "device_tracker",
+        "person",
+        "sensor",
+        "zone"
+    ]
+    
+    private var interfaceController: CPInterfaceController?
+    private var filteredEntities: [HAEntity] = []
+    private var entitiesGridTemplate: EntitiesGridTemplate?
+    private var domainsListTemplate: DomainsListTemplate?
+    private var entitiesStateSubscribeCancelable: HACancellable?
+    private var serverObserver: HACancellable?
+    private var serverId: Identifier<Server>? {
+        didSet {
+            loadEntities()
+        }
+    }
+    
+    let prefs = UserDefaults(suiteName: Constants.AppGroupID)!
+    
+    func loadEntities() {
+        self.domainsListTemplate?.setServerListButton(show: Current.servers.all.count > 1)
+        
+        guard let serverId = serverId, let server = Current.servers.server(for: serverId) else {
+            Current.Log.info("No server available to get entities")
+            filteredEntities.removeAll()
+            self.domainsListTemplate?.entitiesUpdate(updateEntities: filteredEntities)
+            return
+        }
+        
+        guard let allServerEntities = Current.api(for: server).connection.caches.states.value?.all else {
+            Current.Log.info("No entities available from server \(server.info.name)")
+            filteredEntities.removeAll()
+            self.domainsListTemplate?.entitiesUpdate(updateEntities: filteredEntities)
+            interfaceController?.setRootTemplate(self.domainsListTemplate!.getTemplate(), animated: false)
+            return
+        }
+        
+        filteredEntities = getFilteredAndSortEntities(entities: Array(allServerEntities))
+        self.domainsListTemplate?.entitiesUpdate(updateEntities: filteredEntities)
+        if let template = self.domainsListTemplate?.getTemplate() {
+            interfaceController?.setRootTemplate(template, animated: false)
+        }
+    }
+    
+    func getFilteredAndSortEntities(entities: [HAEntity]) -> [HAEntity] {
+        var tmpEntities: [HAEntity] = []
+        
+        for entity in entities where SUPPORTED_DOMAINS.contains(entity.domain) {
+            tmpEntities.append(entity)
+        }
+        return tmpEntities.sorted(by: {$0.getFriendlyState() < $1.getFriendlyState()})
```

Comment:
sorting by this maybe coincidentally achieves the ordering you want, but it seems extremely language specific.

this is also sorting translated strings by non-localized comparisons.